### ISSUE TITLE
removing instances from a project

### ISSUE THREAD

**Issue by ngrosc** [apache/cloudstack#5248]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API, UI,
~~~

##### CLOUDSTACK VERSION

~~~
4.14.1.0
~~~

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### SUMMARY
Removing instances from a project seems not to be possible

##### STEPS TO REPRODUCE

~~~
create a domain
create an account into this domain
create a project into this domain
create an account into this project
create an instance into this project
try to bring the instance from the projects account to the one outside the project
~~~

##### EXPECTED RESULTS

~~~
it should be possible to unassign instances from projects
~~~

##### ACTUAL RESULTS

~~~
as far as I know, there is no way to do this without exporting the instance
~~~

**Comment by DaanHoogland** [comment]
@ngrosc is the account inside the project the same as the one outside?
and if so, isn't this a more generic issue of transferring ownership of VMs?

**Comment by ngrosc** [comment]
Hi Daan

No, they are different. But as you said, it is a sort of ownership transfer..

If this is not possible without hacks, we can export and re-import the vm.. But it would be nice, if the ownership could be changed somehow..

**Comment by DaanHoogland** [comment]
@ngrosc  there was a plan to implement transfer of ownership (https://cwiki.apache.org/confluence/display/CLOUDSTACK/Change+VM+ownership), but I don;t think it ever got anywhere. Another workaround would be to take a snapshot of the VM and to create a template from the snapshot and start a VM from that in the other account. Of course yours may work as well, but only if you are using Vmware as hypervisor, for now.

**Comment by nvazquez** [comment]
@ngrosc @DaanHoogland if you don't mind I'm marking this as an improvement/new feature rather than a bug as by design the owner of the resources within a project is the project itself and not the accounts that belong to the project. It will be interesting to discuss how the unassigning resource from a project should work

**Comment by weizhouapache** [comment]
@nvazquez @ngrosc 
this is not a bug or new feature. 
cloudstack already supports moving vm from an accout to another account or project.

stop the vm and remove all network rules on the vm.
![image](https://user-images.githubusercontent.com/57355700/154667803-d8459440-14d5-4cc5-8af4-93b15e9df1ad.png)

you can move vm to a project
![image](https://user-images.githubusercontent.com/57355700/154665111-dade8afa-70d4-484d-924a-8495ff7f7bef.png)

and move vm from a project to an account
![image](https://user-images.githubusercontent.com/57355700/154668078-398696f6-bc43-4567-a7be-baf4b39616ed.png)

**Comment by weizhouapache** [comment]
@nvazquez 
there is a bug on UI, I will fix it

**Comment by nvazquez** [comment]
@weizhouapache I cannot remove a VM from a project to an account: getting this error:
<img width="400" alt="Screen Shot 2022-02-18 at 14 39 11" src="https://user-images.githubusercontent.com/5295080/154734222-03da248d-498b-46a6-b615-7d26e708b845.png">

Working on a fix

**Comment by nvazquez** [comment]
Fixed on PR #6010
